ratpack-bearer-auth/src/main/java/st/ratpack/auth/springsec/SpringSecCheckTokenProvider.java

```diff
@@ -37,17 +40,24 @@ public SpringSecCheckTokenProvider(
 			.build();
 
 
-		return httpClient.post(uri, rs -> rs
-			.body(body -> body.type(APPLICATION_X_WWW_FORM_URLENCODED.toString()).text("token=" + token))
-			.redirects(0)
-			.headers(headers -> {
-				headers.add(
-					HttpHeaderNames.AUTHORIZATION,
-					buildBasicAuthHeader(config.getUser(), config.getPassword())
-				);
-				headers.add(HttpHeaderNames.ACCEPT, APPLICATION_JSON);
-			})
-		);
+		return httpClient
+			.post(uri, rs -> rs
+				.body(body -> body.type(APPLICATION_X_WWW_FORM_URLENCODED.toString()).text("token=" + token))
+				.redirects(0)
+				.headers(headers -> {
+					headers.add(
+						HttpHeaderNames.AUTHORIZATION,
+						buildBasicAuthHeader(config.getUser(), config.getPassword())
+					);
+					headers.add(HttpHeaderNames.ACCEPT, APPLICATION_JSON);
+				}))
+			.retry(2, (attempt, ex) -> {
+				if (ex instanceof SocketException || ex instanceof PrematureChannelClosureException || ex instanceof SSLException) {
+					return Promise.value(Duration.ofMillis(200 * attempt));
```

Comment:
we should probably handle timeout also

Reply:
The ConnectException actually extends SocketException I learned today, so I think we should be good as long as we don't want a read timeout as well (which maybe we don't if auth oauth service is taking awhile to respond)